meta-balena-common/recipes-core/images/resin-image-flasher.bb

```diff
@@ -2,7 +2,7 @@ SUMMARY = "Resin image flasher"
 IMAGE_LINGUAS = " "
 LICENSE = "Apache-2.0"
 
-inherit core-image image-resin distro_features_check
+inherit core-image image-resin ${@bb.utils.contains('DISTRO_CODENAME', 'dunfell', 'features_check', 'distro_features_check', d)}
```

Comment:
I'd rather see meta-balena-common use `feature_check` and make the corresponding change in the compatibility layers. This avoid distro switches in meta-balena.

Reply:
I hardcoded the fix in common and added compatibility workarounds to warrior and thud. Should I add the compatibility to older layers as well?

Reply:
As long as we have devices types on older layers, yes, see my comment above.